apispec/har.go

```diff
@@ -39,6 +41,67 @@ func processHAR(inboundCol, outboundCol trace.Collector, p string) error {
 	return nil
 }
 
+// ReconstructedTimestamps holds times we can use to fill in
+// a ParsedNetworkTraffic, in such a way that the reported
+// latencies in the Witness will end up the same as in the HAR
+// file.  (Because I couldn't think of a good way to pass
+// the measurements through the to two separate Process
+// calls, without major refactoring.)
+//
+// In the future, we could save these timestamps directly in a
+// custom field, or pass the interval values through.
+type ReconstructedTimestamps struct {
+	RequestStart  time.Time
+	RequestEnd    time.Time
+	ResponseStart time.Time
+	ResponseEnd   time.Time
+}
+
+func advanceTime(current time.Time, ms *float32) time.Time {
+	if ms == nil || *ms <= 0.0 {
+		return current
+	}
+	// convert ms to nanoseconds
+	duration := time.Duration(int64(*ms * 1_000_000.0))
```

Comment:
How about `duration := *ms * time.Millisecond`?

Reply:
That's where I started, it doesn't work.  Only constants can do that.  :(

https://play.golang.org/p/f71AyeNmJoG

./prog.go:9:18: invalid operation: *ms * time.Millisecond (mismatched types float32 and time.Duration)

Reply:
Alas, how disappointing.